Add data frames to Excel workook output. (#11)
Public-Health-Scotland/shetland_pc_cv_ltc: Shetland_PCPIP_Indicators.Rmd

pharma1_180_dt
```

```{r export-all-data-frames, message=FALSE, warning=FALSE}
# Create the named list of data frames and their sheet names
all_data_frames <- list(

>   all_data = all_data,

Reviewer: `all_data` is all the indicators joined together.
```suggestion
```

  monthly_event_type = monthly_event_type,

>   monthly_summary = monthly_summary,

Reviewer: This one, and maybe other dataframes, contain multiple indicators.

Maybe something like.
```r
cv_invites <- monthly_summary |>
  select(PracticeID, census_date, cv_prev, cv_invites)

cv_attend <- monthly_summary |>
  select(PracticeID, census_date, cv_prev, cv_attends)
```

This code will go above the `all_data_frames <- list(` bit, so that they can then be included as separate sheets.

>   pharma_1_monthly = pharma_1_monthly,

Reviewer: Also split pharma1 into 2 tabs - 180-days and 90 days.

  pharmacist_reviews = pharmacist_reviews,

>   quarterly_event_type = quarterly_event_type

Reviewer: Can the sheets have nicer names. You can use `""`.

)

# Create a new workbook
wb <- createWorkbook()

# Add each data frame as a sheet and set column widths to auto

> for (sheet_name in names(all_data_frames)) {
>   addWorksheet(wb, sheet_name)

Reviewer: ```suggestion
  # Make data column names consistent and nice in Excel
  all_data_frames[[sheet_name]] <- janitor::clean_names(
      all_data_frames[[sheet_name]], 
      case = "title",
      abbreviations = c("LTC", "ID")
  )
  
  addWorksheet(wb, sheet_name)
```

  writeData(wb, sheet_name, all_data_frames[[sheet_name]])

>   setColWidths(wb, sheet = sheet_name, cols = 1:ncol(all_data_frames[[sheet_name]]), widths = "auto")

Reviewer: Look at formatting

- [ ] Column widths don't seem to be auto scaling :shrug:
- [ ] Bold the headers
- [ ] Check the `census_date` column - The indicators from `pharma.R` have 'times' on them. Perhaps an `as.Date()` is needed somewhere in the R code? Or get `openxlsx` to format all the' census_date' columns as a nice 'Excel Date'.
- [ ] Maybe rename columns - Go back into the original script, and use `CTRL + SHIFT + F` to find all uses to make sure you're not breaking things.
- [ ] Maybe make a table / add filters
- [ ] Maybe format columns as percentages (might be v tricky).

}

# Save the workbook to the specified path

> file_path <- "/conf/LIST_analytics/Shetland/Primary Care/LTC/data/outputs/Indicators.xlsx"

Reviewer: Let's keep track of the 'date of the data' i.e. what the data goes up to. This is the name of Al's extract(s).

We'll need to define a variable near the top.

```r
data_date <- "May 2025"
```

Something like this can then be added to the top para.

```md
This report is based on data up to the end of `r data_date`.
```

Then we can also use it in the filepath.

```suggestion
file_path <- paste0("/conf/LIST_analytics/Shetland/Primary Care/LTC/data/outputs/Indicators_", data_date, ".xlsx")
```

> saveWorkbook(wb, file_path, overwrite = TRUE)

Reviewer: Do `remotes::install_github("Moohan/phssensitivitylabels")` 
Then add `library(phssensitivitylabels)` at the top of this file.

```suggestion
saveWorkbook(wb, file_path, overwrite = TRUE)

apply_sensitivity_label(file_path, "OFFICIAL_SENSITIVE_VMO")
```
